pynest/nest/lib/hl_api_simulation.py

```diff
@@ -41,6 +42,61 @@ def Simulate(t):
 
 
 @check_stack
+def Run(t):
+    """Simulate the network for t milliseconds.
+
+    Parameters
+    ----------
+    t : float
+        Time to simulate in ms
+
+    Call between Prepare and Cleanup calls, or within an
+    with RunManager: clause
+    Simulate(n) = Prepare() Run(n/m) ... m times ... Cleanup()
+    """
+
+    sps(float(t))
+    sr('ms Run')
+
+
+@check_stack
+def Prepare():
+    """Prepares network before a Run call. Not needed for Simulate.
+    """
+
+    sr('Prepare')
+
+
+@check_stack
+def Cleanup():
+    """Cleans up resources after a Run call. Not needed for Simulate.
+
+    A Prepare is needed after a Cleanup before another Run.
+    """
+    sr('Cleanup')
+
+
+@contextmanager
+def RunManager():
+    """ContextManager for Run.
+
+    Calls Prepare() before a series of Run() calls,
+    and  adds a Cleanup() at end.
+
+    So:
+    with RunManager():
+    	for i in range(10):
+    	    Run()
```

Comment:
Doesn't `Run()` need an argument? I think it would be very useful to explain this context a bit more, so that users (neuroscientist with some Python knowledge) understand how they can use this context manager sensibly. Maybe put a little plot into the loop?